packages/-ember-data/tests/helpers/qunit-asserts/assert-warning.ts

```diff
@@ -0,0 +1,168 @@
+import QUnit from 'qunit';
+import { registerWarnHandler } from '@ember/debug';
+import { checkMatcher } from './check-matcher';
+import RSVP from 'rsvp';
+
+let HAS_REGISTERED = false;
+let WARNINGS_FOR_TEST: FoundWarning[];
+let HANDLED_WARNINGS_FOR_TEST: FoundWarning[];
+
+interface WarningConfig {
+  id: string;
+  count?: number;
+  until?: string;
+  message?: string | RegExp;
+  url?: string;
+}
+interface FoundWarning {
+  message: string;
+  options: {
+    id: string;
+    message?: string;
+    until?: string;
+    url?: string;
+  };
+}
+
+interface AssertSomeResult {
+  result: boolean;
+  actual: { id: string; count: number };
+  expected: { id: string; count: number };
+  message: string;
+}
+interface AssertNoneResult {
+  result: boolean;
+  actual: FoundWarning[];
+  expected: FoundWarning[];
+  message: string;
+}
+
+/**
+ * Returns a qunit assert result object which passes if the given warning
+ * `id` was found *exactly* `count` times.
+ *
+ * Fails if not found or found more or less than `count`.
+ * Fails if `until` not specified
+ * Optionally fails if `until` has been passed.
+ */
+function expectWarning(config: WarningConfig): AssertSomeResult {
+  // TODO optionally throw if `until` is the current version or older than current version
+  let matchedWarnings = WARNINGS_FOR_TEST.filter(warning => {
+    if (!warning.options || !warning.options.id) {
+      throw new Error(`Expected warning to have an id, found: ${warning}`);
```

Comment:
Do we actually need to do this? Doesn't Ember itself error if there is no `id`?

Reply:
See here: 

https://github.com/emberjs/ember.js/blob/129d00b3b4f14826bd9ea4499fd0c8bc7ee5a5a5/packages/%40ember/debug/lib/warn.ts#L105-L106

Reply:
Odd, we've had many without ID's slip past before 🤔

Reply:
Went ahead and removed for now